src/unix/os390-syscalls.c

```diff
@@ -332,3 +332,47 @@ char* mkdtemp(char* path) {
 
   return path;
 }
+
+
+ssize_t os390_readlink(const char* path, char* buf, size_t len) {
+  ssize_t rlen;
+  char* delimiter;
+  char* realpathstr;
+  char old_delim;
+
+  rlen = readlink(path, buf, len);
+
+  /* Straightforward readlink */
+  if (rlen < 2 || strncmp("/$", buf, 2) != 0)
+    return rlen;
+
+  /*
+   * There is a parmlib variable at the beginning
+   * which needs interpretation
+   */
+  delimiter = strchr(buf + 1, '/'); 
+  if (delimiter == NULL)
+    /* No slash at the end */
+    delimiter = strchr(buf + 1, '\0');
+
+  realpathstr = uv__malloc(rlen + 1);
+  if (realpathstr == NULL) {
+    errno = ENOMEM;
+    return -1;
+  }
+
+  /* Read real path of the variable */
+  old_delim = *delimiter;
+  *delimiter = '\0';
+  if (realpath(buf, realpathstr) == NULL) {
+    *delimiter = old_delim;
+    uv__free(realpathstr);
+    return -1;
+  }
+
+  *delimiter = old_delim;
+  rlen = snprintf(buf, len, "%s%s", realpathstr, delimiter);
```

Comment:
Is AIX's `snprintf()` safe to use for overlapping ranges?  I mean that`buf <= delimiter <= buf+len`.